tests/cluster/tests/24-links.tcl

```diff
@@ -8,18 +8,27 @@ test "Cluster should start ok" {
     assert_cluster_state ok
 }
 
+proc number_of_peers {id} {
+    expr [llength [get_cluster_nodes $id]] - 1
+}
+
+proc number_of_links {id} {
+    llength [get_cluster_links $id]
+}
+
 test "Each node has two links with each peer" {
     foreach_redis_id id {
-        # Get number of peers, excluding myself
-        set nodes [get_cluster_nodes $id]
-        set num_peers [expr [llength $nodes] - 1]
+        # Assert that from point of view of each node, there are two links for
+        # each peer. It might take a while for cluster to stabilize so wait up
+        # to 5 seconds.
+        wait_for_condition 50 100 {
+            [number_of_peers $id]*2 == [number_of_links $id]
+        } else {
+            assert_equal [expr [number_of_peers $id]*2] [number_of_links $id]
+        }
 
-        # Get number of links to peers
+        set nodes [get_cluster_nodes $id]
         set links [get_cluster_links $id]
```

Comment:
Assuming the cluster is un stable, Instead of getting `nodes` and `links` again, IMO we should get them while waiting the condition to be met.

```
proc number_of_peers {id nodes} {
    upvar $nodes n
    set n [get_cluster_nodes $id]
    return [expr [llength $n] - 1]
}

proc number_of_links {id links} {
    upvar $links l
    set l [get_cluster_links $id]
    return [llength $l]
}

test "Each node has two links with each peer" {
    foreach_redis_id id {
        set nodes {}
        set links {}
        # Assert that from point of view of each node, there are two links for
        # each peer. It might take a while for cluster to stabilize so wait up
        # to 5 seconds.
        wait_for_condition 50 100 {
            [number_of_peers $id $nodes]*2 == [number_of_links $id $links]
        } else {
            assert_equal [expr [number_of_peers $id $nodes]*2] [number_of_links $id $links]
        }
    
       # Then check if there are two entries in `$links` for each entry in `$nodes`
```

Reply:
Why would the cluster be unstable? I was under the impression that the cluster was still establishing connections as a part of the meeting process, which is why not all links were established. In steady state it shouldn't be unstable.